cache/src/main/scala/org/allenai/common/cache/JsonQueryCache.scala

```diff
@@ -1,10 +1,29 @@
 package org.allenai.common.cache
 
-import com.google.inject.Inject
-import com.google.inject.name.Named
+import org.allenai.common.Config._
+
+import com.typesafe.config.Config
 import redis.clients.jedis.{ Jedis, JedisPool, JedisPoolConfig, Protocol }
 import spray.json._
 
+object JsonQueryCache {
+  /** Factory method for creating a cache instance from config.
+    * The config must have keys for `hostname` and `clientPrefix`. It may also optionally have
+    * keys for `port` and `timeoutMillis`; if not given, these values are set to the Jedis defaults.
+    */
+  def fromConfig[V](config: Config)(implicit jsonFormat: JsonFormat[V]): JsonQueryCache[V] = {
+    // Required fields.
+    val hostname: String = config[String]("hostname")
+    val clientPrefix: String = config[String]("clientPrefix")
+
+    // Optional overrides for Jedis defaults.
+    val port: Int = config.get[Int]("port") getOrElse Protocol.DEFAULT_PORT
+    val timeoutMillis: Int = config.get[Int]("timeoutMillis") getOrElse Protocol.DEFAULT_TIMEOUT
+
+    new JsonQueryCache[V](hostname, port, timeoutMillis, clientPrefix)
```

Comment:
You should call the primary constructor, with a connection pool you create here.